modules/tinymce/src/themes/silver/main/ts/ui/toolbar/button/ToolbarButtons.ts

```diff
@@ -303,140 +304,197 @@ const fetchChoices = (getApi: (comp: AlloyComponent) => Toolbar.ToolbarSplitButt
         )
       )));
 
-// TODO: hookup onSetup and onDestroy
-const renderSplitButton = (spec: Toolbar.ToolbarSplitButton, sharedBackstage: UiFactoryBackstageShared, btnName?: string): SketchSpec => {
-  const tooltipString = Cell<string>(spec.tooltip.getOr(''));
+const makeSplitButtonApi = (tooltipString: Cell<string>, sharedBackstage: UiFactoryBackstageShared, chevronTooltip: string) => (component: AlloyComponent): Toolbar.ToolbarSplitButtonInstanceApi => {
+  const system = component.getSystem();
+  const element = component.element;
+  const isChevron = Class.has(element, 'tox-split-button__chevron');
+  const parentOpt = Traverse.parent(element);
+
+  const mainOpt = isChevron ?
+    parentOpt.bind((parent) =>
+      SelectorFind.descendant(parent, 'button:not(.tox-split-button__chevron)')
+    ).bind((el) => system.getByDom(el).toOptional()) :
+    Optional.some(component);
```

Comment:
I think this always return the first matching descendant in the toolbar group. What if it's the nth button in the toolbar group? 🤔

Reply:
The `SelectorFind.child` method still returns the first matching element. Is there a way to target a unique selector instead?

We should probably include the following sanity tests, especially since these behaviors have changed:
- The dropdown is positioned close to the corresponding button, and
- The "active" state is updated correctly.

The current approach always anchors to the first matching split button and updates its active state.

Reply:
ah yeah, missed `sibling`, this should only return the siblings of the current element

Reply:
so turns out i pushed an archived stash, it is not using selectors anymore, its using `Traverse` and sibling.  `SelectorFind` had the issue you mentioned @shanmen-tiny , but i thought this wasn't really the case because it worked great... a while ago.